From ee15e1f38dc201fa7d63c13aa258b728dce27f4d Mon Sep 17 00:00:00 2001
From: Eric Dumazet <edumazet@google.com>
Date: Thu, 27 Oct 2022 04:06:37 +0000
Subject: [PATCH] kcm: do not sense pfmemalloc status in kcm_sendpage()
Git-commit: ee15e1f38dc201fa7d63c13aa258b728dce27f4d
Patch-mainline: v6.1-rc3
References: git-fixes
Modified-by-SEL: Yes, refreshed due to context changes


Similar to changes done in TCP in blamed commit.
We should not sense pfmemalloc status in sendpage() methods.

Fixes: 326140063946 ("tcp: TX zerocopy should not sense pfmemalloc status")
Signed-off-by: Eric Dumazet <edumazet@google.com>
Link: https://lore.kernel.org/r/20221027040637.1107703-1-edumazet@google.com
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
Signed-off-by: Chenxi Mao <chenxi.mao@suse.com>
---
 net/kcm/kcmsock.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/kcm/kcmsock.c
+++ b/net/kcm/kcmsock.c
@@ -838,7 +838,7 @@ static ssize_t kcm_sendpage(struct socke
 	}
 
 	get_page(page);
-	skb_fill_page_desc(skb, i, page, offset, size);
+	skb_fill_page_desc_noacc(skb, i, page, offset, size);
 	skb_shinfo(skb)->tx_flags |= SKBTX_SHARED_FRAG;
 
 coalesced:
